### PR TITLE
add metrics and cv to refitting in Model.test_estimators

### LINKED ISSUE #751: BUG: When using the method Model.test_estimators with refit=True it's not using the specified metrics.
**Describe the bug**
When using the method Model.test_estimators with refit=True, it's not using the specified metrics. Instead it's using the default metric.

Additionally, wouldn't you except it to be refitting using CV as well if specified in the call?

**To Reproduce**
1) Run the code:
```python
from sklearn.dummy import DummyClassifier
from ml_tooling import Model
from ml_tooling.data.load_demo import load_demo_dataset

data = load_demo_dataset("breast_cancer")
data.create_train_test()

estimators = [DummyClassifier()]

clf, results = Model.test_estimators(data, estimators, metrics="roc_auc", refit=True)

assert clf.result.metrics.name == "accuracy"
clf.result
```
2) Verify that the metric is not roc_auc but accuracy.

**Expected behavior**
I'd expect it to keep evaluating with the previous chosen metrics and I'd expect it to keep using CV if chosen previously.

- [X] I have updated ML Tooling to the latest version
- [X] I have checked if this issue has been reported previously

### PR DESCRIPTION
**Pull request by clausvestergaard**
- [X] closes #751 
- [ ] Tests added
I've added a test for the metrics but I don't really know how to test if the cv is used in refitting.
- [X] Passes flake8
- [ ] Updated CHANGELOG
- [ ] Updated README.md

### CODE REVIEW

**Comment by andersbogsnes** [comment]
Code looks good - remember to update the CHANGELOG and maybe make a note in the docs?
Waiting for discussion on #751 to conclude before accepting

**Comment by andersbogsnes** [comment]
You can test if the scores are cross-validated by looking at the metrics `cross_validated_scores` attribute

**Review by andersbogsnes — APPROVED** [review]
